lib/logstash/outputs/s3.rb

```diff
@@ -79,6 +79,8 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   config_name "s3"
   default :codec, 'line'
 
+  concurrency :single
```

Comment:
yewps, this plugin is in fact in this current state non threadsafe, I've made the plugin run for a few hours to discover that problem.